core/src/player.rs

```diff
@@ -1474,6 +1474,11 @@ impl Player {
 
     #[instrument(level = "debug", skip_all)]
     pub fn render(&mut self) {
+        self.update(|context| {
```

Comment:
I'd consider avoiding always creating a full context and instead pull the check here:
```rs
        let invalidated = self.gc_arena.borrow().mutate(|_, gc_root| {
            gc_root.data.read().stage.invalidated()
        });
        if invalidated {
            self.update(|context| {
                context.stage.broadcast_render(context);
            }
        }
```
But maybe I'm overthinking it.